packages/modules/alpha_ess/inverter.py

```diff
@@ -49,23 +64,23 @@ def __version_factory(self, version: int) -> int:
             else:
                 return 0x00A1
         except Exception as e:
-            log.MainLogger().error("Fehler im Modul "+self.component["name"], e)
+            log.MainLogger().exception("Fehler im Modul "+self.data["config"]["name"])
 
     def __get_power(self, sdmid: int, reg_p: int) -> float:
         try:
             p_reg = self.client.read_binary_registers_to_int(reg_p, 4, sdmid, 32)
-            if p_reg != None:
+            if isinstance(p_reg, (int, float)):
                 if (p_reg < 0):
                     p_reg = p_reg * -1
             time.sleep(0.1)
             p2_reg = self.client.read_binary_registers_to_int(0x041F, 4, sdmid, 32)
             p3_reg = self.client.read_binary_registers_to_int(0x0423, 4, sdmid, 32)
             p4_reg = self.client.read_binary_registers_to_int(0x0427, 4, sdmid, 32)
-            if p2_reg != None and p3_reg != None and p4_reg != None:
+            if isinstance(p2_reg, (int, float)) and isinstance(p3_reg, (int, float)) and isinstance(p4_reg, (int, float)):
                 power = (p_reg + p2_reg + p3_reg + p4_reg) * -1
             else:
-                power = None
+                power = p_reg  # enthält Fehlermeldung
```

Comment:
An dem Kommentar sieht man nochmal wie schwierig das ist mit der Fehlermeldung. Man muss hier aktiv immer sich bewusst sein, dass `power` ja auch eine Fehlermeldung sein könnte. (Was übrigens heißt dass das Type-hinting an der Methode auch wieder falsch ist. Es ist nicht unbedingt ein `float`).

Und obwohl sich der Autor dem offensichtlich bewusst war, ist es immernoch nicht richtig. Denn theoretisch ist es möglich dass `p_reg` völlig in Ordnung ist und kein Fehler enthält, aber p2_reg einen Fehler enthält.

Wieso eigentlich `p_reg`, `p2_reg` etc?
1. Wäre es konsequenter dann auch mit `p2_reg` anzufangen.
2. Was heißt eigentlich `p2`? Ich dachte erst vielleicht "phase 2", aber es gibt ziemlich sicher keine "phase 4".
3. Ein Array würde hier einiges vereinfachen.